bootable-jar/WFLY-13305_Bootable_Jar.adoc

```diff
@@ -0,0 +1,326 @@
+This is similar to the _--properties=<properties file>_ option that you can provide to JBoss CLI command line. 
+This http://www.mastertheboss.com/jboss-server/jboss-script/using-properties-in-cli-scripts[article] covers the feature.
+
+The https://github.com/wildfly-extras/wildfly-jar-maven-plugin/releases/download/2.0.0.Alpha4/index.html[community documentation] contains details on how to setup CLI scripts and CLI properties file.
+
+===== User application
+
+The WAR or JAR main artifact the Maven project is building is deployed and zipped along with the server. 
+The plugin option root-url-path=true|false (true by default) allows to rename the war file to ROOT.war 
+(doesn’t apply to other packaging types). 
+
+NB: Application deployment doesn't rely on the deployment scanner. The scanner is not required, it is even suggested to exclude it, useless in a bootable JAR context.
+For an example of how to exclude the _deployment-scanner_ Galleon layer, you can check https://github.com/wildfly-extras/wildfly-jar-maven-plugin/blob/master/examples/jaxrs/pom.xml#L35[this example]
+
+In case the bootable is an “hollow JAR”, no deployment is present in the JAR.
+
+===== Hollow JAR
+
+No WAR/JAR/EAR file is copied into the server deployments. This is controlled by a plugin option.
+The hollow JAR doesn’t require the deployment-scanner to be provisioned in order for the deployment to be taken into account at runtime time. At runtime, deployment is copied to content dir, xml configuration is updated with the deployment.
+
+
+==== Execute the bootable JAR
+
+This is done thanks to the command: java -jar <JAR name>.jar [arguments]
+The detailed execution steps are:
+
+* Unzip the server to a directory
+* Handle arguments
+* Start the server
+* Wait for server end 
+
+Alternatively the Maven plugin can be used to run/shutdown the server from Maven wildfly-jar:run|start|shutdown.
+
+==== Configure the Bootable JAR at runtime
+
+The bootable JAR fully relies on the server configuration capabilities. It is not expected to define a new way to configure the server.
+The arguments passed to the bootable JAR are composed of bootable specific arguments and server arguments.
+
+Calling  java <JVM and system properties> -jar <bootable JAR name>.jar --help dumps the available arguments.
+
+==== Log manager
+
+The bootable JAR depends on JBoss log manager. Logging is configured thanks to the logging subsystem.
+
+==== Shutting down the Bootable JAR
+
+The bootable JAR process can be shutdown in the following ways:
+
+* Signal handling
+* :shutdown management operation with the following limitations:
+** The restart option is not supported and would be ignored by the server. The process will exit in all cases.
+
+
+==== Read Only standalone.xml
+
+The changes made to the management model are not reflected in the standalone.xml. 
+The server is internally started with --read-only-server-config=standalone.xml
+
```

Comment:
A question that popped several times app lately when discussing this proposal was: "So is the user able to choose another `standalone-*.xml` for server config?". Although I think it is pretty straightforward that the configuration is based on defined Galleon layers and the config file is built from it, I think it could be useful for the reader to have this fact mentioned as the cause for it to be read only, here. WDYT?

Reply:
I explained why it is readonly and added the fact that we can't replace the standalone.xml with a custom one in the "Non Requirements"